File: livingrimoire start here/LivinGrimoire java/LivinGrimoire/Algorithm.java
package LivinGrimoire;

import java.util.ArrayList;
import java.util.Arrays;

// a step-by-step plan to achieve a goal
public class Algorithm {
    private ArrayList<AlgPart> algParts = new ArrayList<>();

    public Algorithm(ArrayList<AlgPart> algParts) {
        this.algParts = algParts;
    }
    public Algorithm(AlgPart... algParts) {
        this.algParts = new ArrayList<>(Arrays.asList(algParts));
    }

    public ArrayList<AlgPart> getAlgParts() {
        return algParts;
    }

    public int getSize() {
        return algParts.size();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/livinGrimoire/Algorithm.kt
package livinGrimoire

import java.util.*

// a step-by-step plan to achieve a goal
class Algorithm(val algParts: ArrayList<AlgPart>) {
    // Secondary constructor for varargs
    constructor(vararg algParts: AlgPart) : this(ArrayList(algParts.toList()))

    // Property to get the size of algParts
    val size: Int
        get() = algParts.size
}


